src/nano_id/core.cljc

```diff
@@ -1,28 +1,54 @@
 (ns nano-id.core
-  (:require [nano-id.random :as rnd]))
+  (:require [nano-id.random :as rnd])
+  #?(:clj (:import nano_id.NanoID)))
 
 (def alphabet
   (mapv str "_-0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"))
 
 (defn nano-id
   "Secure random ID generator.
-  Generates IDs of the specified `size`, it's 21 by default."
-  ([] (nano-id 21))
+  Generates IDs of the specified size, 21 by default."
+  ([]
+   #?(:clj  (NanoID/nanoID)
+      :cljs (nano-id 21)))
   ([size]
-   #?(:clj
-      (let [mask 0x3f
-            bytes (clojure.lang.RT/iter ^bytes (rnd/random-bytes size))]
-        (loop [id (StringBuilder.)]
-          (if (.hasNext bytes)
-            (recur (let [ch (alphabet (bit-and (.next bytes) mask))]
-                     (.append id ch)))
-            (str id))))
-      :cljs
-      (let [mask 0x3f]
-        (loop [bytes (rnd/random-bytes size)
-               id ""]
-             (if bytes
-               (recur (next bytes)
-                      (let [ch (nth alphabet (bit-and (first bytes) mask))]
-                        (str id ch)))
-               id))))))
+   #?(:clj  (NanoID/nanoID size)
+      :cljs (let [mask 0x3f]
+              (loop [bytes (rnd/random-bytes size)
+                     id ""]
+                   (if bytes
+                     (recur (next bytes)
+                            (->> (first bytes)
+                                 (bit-and mask)
+                                 alphabet
+                                 (str id)))
+                     (str id)))))))
```

Comment:
```suggestion
                     id))))))
```

Reply:
Nice catch, forgot about it when separated clj and cljs versions.